File: sem4/mpp/mpp-proiect-csharp-vreauladudu/domain/Race.cs
namespace mpp_proiect_csharp_vreauladudu.domain
{
    enum MotorcycleCategory
    {
        SLOW_125CMC,
        AVERAGE_250CMC,
        FAST_600CMC,
        SUPERFAST_1000CMC,

        MAX_CAPACITY
    }

    internal class Race : Entity<int>
    {
        public MotorcycleCategory motorcycleCategory { get; set; }
        public List<Participant>? participants { get; set; }
    }
}

File: sem4/mpp/mpp-proiect-java-vreauladudu/src/main/java/moto/domain/Race.java
package moto.domain;

import java.util.List;

public class Race extends Entity<Integer>
{
    private MotorcycleCategory motorcycleCategory;
    private List<Participant> participants;

    public MotorcycleCategory getMotorcycleCategory()
    {
        return this.motorcycleCategory;
    }

    public void setMotorcycleCategory(MotorcycleCategory motorcycleCategory)
    {
        this.motorcycleCategory = motorcycleCategory;
    }
}